sematic/db/models/external_resource.py

```diff
@@ -0,0 +1,196 @@
+# Standard Library
+import datetime
+from typing import Any, Dict, Tuple, Type, Union
+
+# Third-party
+from sqlalchemy import Column, types
+from sqlalchemy.orm import validates
+
+# Sematic
+from sematic.db.models.base import Base
+from sematic.db.models.json_encodable_mixin import JSONEncodableMixin
+from sematic.external_resource import ExternalResource as ExternalResourceDataclass
+from sematic.external_resource import ManagedBy, ResourceState
+from sematic.types.serialization import (
+    type_from_json_encodable,
+    type_to_json_encodable,
+    value_from_json_encodable,
+    value_to_json_encodable,
+)
+
+TypeSerialization = Dict[str, Any]
+
+
+class ExternalResource(Base, JSONEncodableMixin):
+    """A DB record for an ExternalResource (dataclass) and its history.
+
+    For the remainder of this docstring, ExternalResource will correspond
+    to the non-orm representation (aka the dataclass representation).
+
+    Attributes
+    ----------
+    id:
+        The unique id of the external resource
+    resource_state:
+        The current state of the resource
+    managed_by:
+        Whether the resource is managed locally, remotely, or its management
+        state is not known.
+    status_message:
+        The most recent status message for the resource
+    last_updated_epoch_seconds:
+        The time that the resource was last updated, expressed as epoch seconds
```

Comment:
Please add an explicit explanation that the `created_at` and `updated_at` fields refer to the DB operations times, and this refers to the time when the resource state was updated based on external infrastructure communication.

Reply:
oops! Added this already, but to the wrong PR. Will transfer here.